include/i2s.h

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2017 Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for the I2S (Inter-IC Sound) bus drivers.
+ */
+
+#ifndef __I2S_H__
+#define __I2S_H__
+
+/**
+ * @defgroup i2s_interface I2S Interface
+ * @ingroup io_interfaces
+ * @brief I2S (Inter-IC Sound) Interface
+ *
+ * The I2S API provides support for the standard I2S interface standard as well
+ * as common non-standard extensions such as PCM Short/Long Frame Sync,
+ * Left/Right Justified Data Format.
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * The following #defines are used to configure the I2S controller.
+ */
+
+
+typedef u8_t i2s_fmt_t;
```

Comment:
I would get rid of such specific typedef, it's just fine to keep u8_t in the config struct and documenting it so it should be populated with the relevant definitions

In device drivers api usually it has been avoided to create such specific types.